ErikBagratyan/File_Problem

```diff
@@ -0,0 +1,43 @@
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <algorithm>
+#include <map>
+#include <vector>
+
+using namespace std;
+
+int main()
+{
+	map<string, int> mp;
+	fstream fin;
+	fin.open("war+peace.txt", fstream::in | fstream::out | fstream::app);
+	if (!fin.is_open())
+	{
+		cout << "Error OPEN File\n";
+	}
+	else {
+		char ch;
+		string str = "";
+		while (!fin.eof())
+		{
+			str = "";
+			while (fin.get(ch) && ch != ' ' && ch != '\n') str += ch;
+			if(str != "") mp[str]++;
+		}
+	}
+	fin.close();
+
+	vector<pair<int, string>> vec(mp.size());
+	for (map<string, int>::const_iterator it = mp.begin(); it != mp.end(); ++it)
+	{
+		vec.push_back({ it->second,it->first });
+	}
+	sort(vec.begin(), vec.end());
```

Comment:
can you make `std::sort` to sort in descending order?

Reply:
yes then the function should accept the 3rd parameter which is a bool type function and gives information on how we want to sort the data,         
bool myFunction(const pair<int, string>& num1, const pair<int, string>& num2)
{
	return (num1 > num2);
}

Reply:
Good, so you can then write `sort(vec.begin(), vec.end(), myFunction)`. However,

1. fix your code in `myFunction` to not compare `num1` and `num2` but `num1.first` and `num2.first` (`return num1.first > num2.first)`.
2. since you have the freedom to choose how to compare 2 pairs, you don't need to push pairs to vector reversed. `vec.push_back({ it->second,it->first });` can simply be `vec.push_back(*it)`. Anyway, it really good that you came up with that idea of reversing the pair before pushing to vector.

Make those changes and submit a new version.